### PR TITLE
fix(Types):Fixed the types of whileElement to be aligned with the documentation …

### PR DESCRIPTION
**Pull request by gosha212**
…and the reality

## Description

- This pull request addresses the issue described here: [#3662](https://github.com/wix/Detox/issues/3662)

In this pull request, I have removed the unsupported actions from waitFor().whileElement()

---

> _For features/enhancements:_
 - [ ] I have added/updated the relevant references in the [documentation](https://github.com/wix/Detox/tree/master/docs) files.

> _For API changes:_
 - [ ] I have made the necessary changes in the [types index](https://github.com/wix/Detox/blob/master/detox/index.d.ts) file.

### CODE REVIEW

**Comment by asafkorem** [comment]
Let's emphasize this on our docs (only scroll is allowed)

File: detox/detox.d.ts [hunk 1]
```diff
@@ -1279,12 +1279,31 @@ declare global {
...
+          /**
+           * Scrolls a given amount of pixels in the provided direction, starting from the provided start positions.
+           * @param pixels - independent device pixels
+           * @param direction - left/right/up/down
+           * @param startPositionX - the X starting scroll position, in percentage; valid input: `[0.0, 1.0]`, `NaN`; default: `NaN`—choose the best value automatically
+           * @param startPositionY - the Y starting scroll position, in percentage; valid input: `[0.0, 1.0]`, `NaN`; default: `NaN`—choose the best value automatically
+           * @example await element(by.id('scrollView')).scroll(100, 'down', NaN, 0.85);
+           * @example await element(by.id('scrollView')).scroll(100, 'up');
+           */
+          scroll(
```
Comment (noomorph) [inline detox/detox.d.ts:1298]: Please add an extra test to:

https://github.com/wix/Detox/blob/feat/refine_whileElement_api/detox/test/types/detox-global-tests.ts#L93

Invoke the `waitFor(...).scroll(pixels, direction, startX, startY`, in addition to `(pixels, directions)`.

Besides, you can add one more test like `// @ts-expect-error` to ensure that `waitFor(...)` does not have `tap` or some other common method.

**Review by noomorph — APPROVED** [review]
Looks good but I strongly recommend to add a couple of typing tests, still.

**Comment by asafkorem** [comment]
Released with Detox [20.21.0](https://github.com/wix/Detox/releases/tag/20.21.0).

**Pull request merged**